src/components/expanded-state/asset/ChartExpandedState.js

```diff
@@ -265,11 +264,10 @@ export default function ChartExpandedState({ asset }) {
     ),
   });
 
-  const uniswapAssetsInWallet = useUniswapAssetsInWallet();
+  const { sortedAssets } = useSelector(sortAssetsByNativeAmountSelector);
   const showSwapButton = useMemo(
-    () =>
-      !isL2 && find(uniswapAssetsInWallet, ['address', assetWithPrice.address]),
-    [assetWithPrice.address, isL2, uniswapAssetsInWallet]
+    () => !isL2 && find(sortedAssets, ['address', assetWithPrice.address]),
```

Comment:
use `uniswapAssetsInWallet` but now you can use the address as key instead of doing `find` on an array as it is now a map